internal/policy/source/source.go

```diff
@@ -65,12 +67,30 @@ type PolicyUrl struct {
 	Kind policyKind
 }
 
+// downloadCache is a concurrent map used to cache downloaded files.
+var downloadCache sync.Map
+
 // GetPolicies clones the repository for a given PolicyUrl
 func (p *PolicyUrl) GetPolicy(ctx context.Context, workDir string, showMsg bool) (string, error) {
 	sourceUrl := p.PolicyUrl()
 
 	dest := uniqueDestination(workDir, p.Subdir(), sourceUrl)
 
+	// Check if the policy has already been downloaded.
+	// If it has try to create a symlink. If that passes return the symlink path.
+	// If the symlink creation fails, fall back to downloading the policy again.
+	if d, loaded := downloadCache.LoadOrStore(sourceUrl, dest); loaded {
+		log.Debugf("Download cache hit: %s", sourceUrl)
+		fs := new(afero.OsFs)
```

Comment:
Doing this helps with tests (using [utils](https://github.com/enterprise-contract/ec-cli/blob/01b6c39ad710b0c6d8e3a70885557c36343e0db5/internal/utils/helpers.go#L94)):

```suggestion
		fs := utils.FS(ctx)
```